load_test.py

```diff
@@ -14,22 +14,43 @@
 import utils
 
 
-def run_main_process(concurrency, duration, dataset, dataset_q, stop_q):
+def run_main_process(rps, duration, dataset, request_q, stop_q):
     """Run the main process."""
     logging.info("Test from main process")
 
-    # Initialize the dataset_queue with 4*concurrency requests
+    start_time = time.time()
+    end_time = start_time + duration
+    if rps is not None:
+        main_loop_rps_mode(dataset, request_q, rps, start_time, end_time)
+    else:
+        main_loop_concurrency_mode(dataset, request_q, start_time, end_time)
+
+    logging.info("Timer ended, stopping processes")
+
+    # Signal users to stop sending requests
+    stop_q.put(None)
+
+    # Empty the dataset queue
+    while not request_q.empty():
+        logging.debug("Removing element from request_q")
+        request_q.get()
+
+    return
```

Comment:
Drop this return?

```suggestion
```

Reply:
It doesn't but I wonder if adding a dedicated try-catch exception block in this function worth it. We currently catch all the cascade exceptions with the generic `Exception` class in the main function but it's probably not the cleanest way to handle the exception IMO.

Not suggesting this should be addressed in this PR but a follow-up PR to cleanup our exception handling might be good.